Two superpmi.py tweaks

BruceForstall opened the pull request:
1. Add superpmi.py command line to the log file. This makes it easier to reproduce the run when you only have a log file (such as from the CI).
2. When outputting assertions in replay, prefix each assertion with a string that makes it easier to find in a large log file, and number the assertions that are output.

BruceForstall committed:
Two superpmi.py tweaks

1. Add superpmi.py command line to the log file. This makes it easier to
   reproduce the run when you only have a log file (such as from the
   CI).
2. When outputting assertions in replay, prefix each assertion with a
   string that makes it easier to find in a large log file, and
   number the assertions that are output.

BruceForstall commented:
@jakobbotsch @dotnet/jit-contrib PTAL

jakobbotsch commented on src/coreclr/scripts/superpmi.py:
```diff
@@ -1618,8 +1618,10 @@ def report_replay_asserts(asserts, output_mch_file):
+            logging.info("=== Assertion #%s: %s", assertion_num, assertion_key)
             # Sort the values by increasing il size
             sorted_instances = sorted(assertion_value, key=lambda d: d['il'])
             for instance in sorted_instances:
```
Can we also limit the amount of examples printed here to the smallest N number of assertions? I sometimes end up with a console full of assertions that I have to scroll up through.

jakobbotsch reviewed (approved):
LGTM with a suggestion

BruceForstall committed:
Code review feedback

1. Add more detail about the assert, including how many instances
2. Limit the number of instances summarized to an arbitrary number of 25
